cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -182,7 +183,15 @@ public void start() {
       LOG.error("Could not disable caching of URLJarFiles. This may lead to 'too many open files` exception.", e);
     }
 
-    ConfigurationLogger.logImportantConfig(cConf);
+    startupChecker.startAndWait();
+    if (!startupChecker.runSanityChecks()) {
+      LOG.error("Errors while starting up master. " +
+                  "Please check the logs, address all errors, then start the master again. " +
+                  "Startup checks can also be disabled by setting {} to false in cdap-site.xml",
+                Constants.STARTUP_CHECKS_ENABLED);
+      Runtime.getRuntime().exit(1);
```

Comment:
did this because I was stopping the thread in stop().  That part could be changed though and we could just exit here.